mathics/builtin/specialfns/bessel.py

```diff
@@ -341,14 +341,14 @@ class BesselI(_Bessel):
      = -Graphics-
     """
 
-    rules = {
-        "Derivative[0, 1][BesselI]": "((BesselI[-1 + #1, #2] + BesselI[1 + #1, #2])/2)&",
-    }
-
     mpmath_name = "besseli"
     rules = {
         "BesselI[Undefined, x_]": "Undefined",
         "BesselI[y_, Undefined]": "Undefined",
+        # FIXME: thes are not respected. Why?
+        "BesselI[x_, -I Infinity]": "0",
```

Comment:
What about using `DirectedInfinity[I]`?

Reply:
Doesn't change things. Ditto for `DirectedInfiity[-I]`  and `-DirectedInfity[i]` and `-ComplexInfiity`. 

I wonder if Sympy handling is skipping rule application. I see  in other situations too.

Reply:
What does TraceEvaluation show?

Reply:
```
In[2]:= TraceEvaluation[BesselI[x, -I Infinity]]
    Evaluating: System`BesselI[Global`x, System`Times[-1, System`I, System`Infinity]]
      Evaluating: System`BesselI
      Evaluating: Global`x
      Evaluating: System`Times[-1, System`I, System`Infinity]
        Evaluating: System`Times
        Evaluating: System`I
        -> I
        Evaluating: System`Infinity
        -> System`DirectedInfinity[1]
        Evaluating: System`DirectedInfinity[1]
          Evaluating: System`DirectedInfinity
          Evaluating: System`Unequal[System`N[System`Abs[1]], 1]
            Evaluating: System`Unequal
            Evaluating: System`N[System`Abs[1]]
              Evaluating: System`N
              Evaluating: System`Abs[1]
                Evaluating: System`Abs
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1.0]
              Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
                Evaluating: System`Function
            -> System`Not[System`ExactNumberQ[1.0]]
              Evaluating: System`Not
              Evaluating: System`ExactNumberQ[1.0]
                Evaluating: System`ExactNumberQ
              -> System`False
            -> System`True
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1]
              Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
                Evaluating: System`Function
            -> System`Not[System`ExactNumberQ[1]]
              Evaluating: System`Not
              Evaluating: System`ExactNumberQ[1]
                Evaluating: System`ExactNumberQ
              -> System`True
            -> System`False
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1.0]
              Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
                Evaluating: System`Function
            -> System`Not[System`ExactNumberQ[1.0]]
              Evaluating: System`Not
              Evaluating: System`ExactNumberQ[1.0]
                Evaluating: System`ExactNumberQ
              -> System`False
            -> System`True
            Evaluating: System`ExactNumberQ[1.0]
              Evaluating: System`ExactNumberQ
            -> System`False
            Evaluating: System`ExactNumberQ[1]
              Evaluating: System`ExactNumberQ
            -> System`True
            Evaluating: System`$MaxExtraPrecision
          -> System`False
      -> System`DirectedInfinity[-I]
        Evaluating: System`DirectedInfinity
        Evaluating: System`Unequal[System`N[System`Abs[-I]], 1]
          Evaluating: System`Unequal
          Evaluating: System`N[System`Abs[-I]]
            Evaluating: System`N
            Evaluating: System`Abs[-I]
              Evaluating: System`Abs
          Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1.0]
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
              Evaluating: System`Function
          -> System`Not[System`ExactNumberQ[1.0]]
            Evaluating: System`Not
            Evaluating: System`ExactNumberQ[1.0]
              Evaluating: System`ExactNumberQ
            -> System`False
          -> System`True
          Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1]
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
              Evaluating: System`Function
          -> System`Not[System`ExactNumberQ[1]]
            Evaluating: System`Not
            Evaluating: System`ExactNumberQ[1]
              Evaluating: System`ExactNumberQ
            -> System`True
          -> System`False
          Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]][1.0]
            Evaluating: System`Function[System`Not[System`ExactNumberQ[System`Slot[1]]]]
              Evaluating: System`Function
          -> System`Not[System`ExactNumberQ[1.0]]
            Evaluating: System`Not
            Evaluating: System`ExactNumberQ[1.0]
              Evaluating: System`ExactNumberQ
            -> System`False
          -> System`True
          Evaluating: System`ExactNumberQ[1.0]
            Evaluating: System`ExactNumberQ
          -> System`False
          Evaluating: System`ExactNumberQ[1]
            Evaluating: System`ExactNumberQ
          -> System`True
          Evaluating: System`$MaxExtraPrecision
        -> System`False
      Evaluating: Global`x
      Evaluating: System`ComplexInfinity
      -> System`DirectedInfinity[]
      Evaluating: System`DirectedInfinity[]
        Evaluating: System`DirectedInfinity
    -> System`BesselI[Global`x, System`DirectedInfinity[]]
      Evaluating: System`BesselI
      Evaluating: Global`x
      Evaluating: System`DirectedInfinity[]
      Evaluating: Global`x
      Evaluating: System`ComplexInfinity
      -> System`DirectedInfinity[]
      Evaluating: System`DirectedInfinity[]
        Evaluating: System`DirectedInfinity
Out[2]= BesselI[x, ComplexInfinity]
```

Reply:
Notice that most of the evaluation is about convert `I infinity` into `ComplexInfinity`, which is also wrong. That are the rules to be fixed.

Reply:
Ok. I will leave it for you or someone else (or maybe me in the future) to fix.